compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -955,85 +965,89 @@ trait Implicits { self: Typer =>
         }
       }
 
-      formal.member(tpnme.MirroredType).info match {
-        case TypeBounds(mirroredType, _) => mirrorFor(mirroredType)
-        case other => EmptyTree
-      }
+      if (!checkFormal(formal)) EmptyTree
+      else
+        formal.member(tpnme.MirroredType).info match {
+          case TypeBounds(mirroredType, _) if checkFormal(formal) => mirrorFor(mirroredType)
```

Comment:
`checkFormal(formal)` is always true here

Reply:
I don't think that can be the case. It being false for `formal` which include def/val definitions is what's responsible for the (intended) neg test failures.

Reply:
But it just got tested in line 968, so to reach the else part, it must be true.

Reply:
Oh! Yes, sorry ... I completely missed the redundant guard there. Removed.